plugin/storage/cassandra/spanstore/writer.go

```diff
@@ -125,6 +134,27 @@ func (s *SpanWriter) Close() error {
 // WriteSpan saves the span into Cassandra
 func (s *SpanWriter) WriteSpan(span *model.Span) error {
 	ds := dbmodel.FromDomain(span)
+	switch s.storageMode {
+	case storeOnly:
+		if err := s.writeSpan(span, ds); err != nil {
+			return err
+		}
+	case indexOnly:
+		if err := s.writeIndexes(span, ds); err != nil {
+			return err
+		}
+	default:
+		if err := s.writeSpan(span, ds); err != nil {
+			return err
+		}
+		if err := s.writeIndexes(span, ds); err != nil {
+			return err
+		}
```

Comment:
duplication of code is a sign of design problem. You could use bitmasks instead of iota-based constants

Reply:
That was what I was doing previously...

Reply:
I like the previous version better, why did you change it? We might have gotten the wires crossed. If your original version of the constants was bitmap-based, then it wasn't clear from the code - I would just declare two constants, `saveSpanFlag 1` and `indexSpanFlag 2`, and add a comment that they can be combined in a bitmap (also name the struct variable `somethingFlags`)

Reply:
I thought that we wanted the default of indexing and storing to have an ordinal of 0.

Reply:
if we're not using the sequential enum then it doesn't matter.